swim/member_predicate.go

```diff
@@ -0,0 +1,43 @@
+package swim
+
+// MemberPredicate is a function that tests if a Member satisfies a condition.
+// It is advised to use exported functions on Member instead of its exported
+// fields in case we want to extract the functionality of Member to an Interface
+// in the future. This is likely to happen if we pursue plugable membership.
+type MemberPredicate func(member Member) bool
+
+// TestMember can take multiple predicates and test them against a member
+// returning if the member satisfies all the predicates. This means that if one
+// test fails it will stop executing and return with false.
+func TestMember(member Member, predicates ...MemberPredicate) bool {
```

Comment:
I'm open for a suggestion. I prefixed it with `Test` here because it tests if the member matches the predicates. However I share your sentiment about the `Test` prefix being very specific for tests in golang.